src/serializers/shared.rs

```diff
@@ -230,6 +235,35 @@ impl CombinedSerializer {
         Self::find_serializer(type_, schema, config, definitions)
     }
 
+    fn maybe_wrap_in_polymorphism_trampoline(
+        serializer: Arc<CombinedSerializer>,
+        schema: &Bound<'_, PyDict>,
+    ) -> PyResult<Arc<CombinedSerializer>> {
+        let py = schema.py();
+        let type_: Bound<'_, PyString> = schema.get_as_req(intern!(py, "type"))?;
+        let type_ = type_.to_str()?;
+
+        if type_ == "model" || type_ == "dataclass" {
```

Comment:
```suggestion
        // Note: it could make sense to generalize this behavior for any type that may have subclasses,
        // but apart from models and dataclasses, that would be for arbitrary types where custom serialization
        // has to be defined already. 
        if type_ == "model" || type_ == "dataclass" {
```